core/src/test/java/com/graphhopper/routing/PathTest.java

```diff
@@ -309,20 +310,60 @@ public void testCalcDetails() {
         Path p = new Dijkstra(g, new ShortestWeighting(encoder), TraversalMode.NODE_BASED).calcPath(1, 5);
         assertTrue(p.isFound());
 
-        Map<String, List<PathDetail>> details = p.calcDetails(new PathDetailsBuilderFactory(Arrays.asList(new String[]{Parameters.DETAILS.AVERAGE_SPEED}), encoder), 0);
-        assertTrue(details.size() == 1);
-        List<PathDetail> detailList = details.get(Parameters.DETAILS.AVERAGE_SPEED);
-        assertEquals(4, detailList.size());
-        assertEquals(45.0, detailList.get(0).getValue());
-        assertEquals(90.0, detailList.get(1).getValue());
-        assertEquals(10.0, detailList.get(2).getValue());
-        assertEquals(45.0, detailList.get(3).getValue());
-
-        assertEquals(0, detailList.get(0).getFirst());
-        assertEquals(1, detailList.get(1).getFirst());
-        assertEquals(2, detailList.get(2).getFirst());
-        assertEquals(3, detailList.get(3).getFirst());
-        assertEquals(4, detailList.get(3).getLast());
+        Map<String, List<PathDetail>> details = p.calcDetails(new PathDetailsBuilderFactory(Arrays.asList(new String[]{AVERAGE_SPEED, STREET_NAME, EDGE_ID, TIME}), encoder, new FastestWeighting(encoder)), 0);
+        assertTrue(details.size() == 4);
+        List<PathDetail> averageSpeedDetails = details.get(AVERAGE_SPEED);
+        assertEquals(4, averageSpeedDetails.size());
+        assertEquals(45.0, averageSpeedDetails.get(0).getValue());
+        assertEquals(90.0, averageSpeedDetails.get(1).getValue());
+        assertEquals(10.0, averageSpeedDetails.get(2).getValue());
+        assertEquals(45.0, averageSpeedDetails.get(3).getValue());
+
+        assertEquals(0, averageSpeedDetails.get(0).getFirst());
+        assertEquals(1, averageSpeedDetails.get(1).getFirst());
+        assertEquals(2, averageSpeedDetails.get(2).getFirst());
+        assertEquals(3, averageSpeedDetails.get(3).getFirst());
+        assertEquals(4, averageSpeedDetails.get(3).getLast());
+
+        List<PathDetail> streetNameDetails = details.get(STREET_NAME);
```

Comment:
We should have different test methods for the different path details (and maybe reuse the common parts)